src/lib/spinel/radio_spinel.hpp

```diff
@@ -1061,26 +1222,35 @@ template <typename InterfaceType> class RadioSpinel
     otRadioFrame  mAckRadioFrame;
     otRadioFrame *mTransmitFrame; ///< Points to the frame to send
 
-    otExtAddress mExtendedAddress;
-    uint16_t     mShortAddress;
-    uint16_t     mPanId;
-    otRadioCaps  mRadioCaps;
-    uint8_t      mChannel;
-    int8_t       mRxSensitivity;
-    otError      mTxError;
-    char         mVersion[kVersionStringSize];
-    otExtAddress mIeeeEui64;
+    otExtAddress        mExtendedAddress;
+    uint16_t            mShortAddress;
+    uint16_t            mPanId;
+    uint8_t             mChannel;
+    int8_t              mRxSensitivity;
+    otError             mTxError;
+    static char         sVersion[kVersionStringSize];
+    static otExtAddress sIeeeEui64;
+    static otRadioCaps  sRadioCaps;
 
     State mState;
-    bool  mIsPromiscuous : 1;     ///< Promiscuous mode.
-    bool  mIsReady : 1;           ///< NCP ready.
-    bool  mSupportsLogStream : 1; ///< RCP supports `LOG_STREAM` property with OpenThread log meta-data format.
-    bool  mIsTimeSynced : 1;      ///< Host has calculated the time difference between host and RCP.
+    bool  mIsPromiscuous : 1; ///< Promiscuous mode.
+    bool  mIsTimeSynced : 1;  ///< Host has calculated the time difference between host and RCP.
+
+    static bool sIsReady;           ///< NCP ready.
+    static bool sSupportsLogStream; ///< RCP supports `LOG_STREAM` property with OpenThread log meta-data format.
 
 #if OPENTHREAD_SPINEL_CONFIG_RCP_RESTORATION_MAX_COUNT > 0
 
+    enum
+    {
+        kRcpFailureNone,
+        kRcpFailureTimeout,
+        kRcpFailureUnexpectedReset,
+    };
```

Comment:
Just fyi, we removed these RCP recovery changes in a later commit in my PR. Not sure if it is not needed for this PR, though. We will have to be a little careful since it would affect non-multipan case as well.